packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+        if (this.columnConfig) {
+            this.text = formatFileSize(
+                this.columnConfig.formatter, // instance of Intl.NumberFormat
+                this.unitStrings, // either ['byte', 'bytes', 'KB', ...] or ['byte', 'bytes', 'KiB', ...]
+                this.columnConfig.unitType,
+                this.cellRecord?.value
+            );
+        } else {
+            this.text = '';
+        }
+    }
+}
+
+// Shared function also used by group header view
+function formatFileSize(
+    formatter: Intl.NumberFormat,
+    unitStrings: string[],
+    unitType: FileSizeUnitType,
+    byteCount: number
+): string {
+    let currentSize = byteCount;
+    let unitIndex = 0;
+    if (currentSize !== 1) {
+        const divisor = unitType === FileSizeUnitType.binary ? 1024 : 1000;
+        while (currentSize >= divisor && unitIndex < unitStrings.length) {
+            currentSize /= divisor;
+            unitIndex += 1;
+        }
+    }
+
+    return `${formatter.format(currentSize)} ${unitStrings[unitIndex]}`;
+}
+```
+
+Group header view will extend `TableColumnTextGroupHeaderViewBase` and use its template.
+
+### States
+
+N/A
+
+### Accessibility
+
+Accessibility of the cells rendered using the `nimble-table-column-file-size` are handled via the [`nimble-table-cell`](https://github.com/ni/nimble/blob/f663c38741e731bef91aa58e8fb2d1cec653b679/packages/nimble-components/src/table/components/cell/template.ts#L6) which has a `role` of [`cell`](https://w3c.github.io/aria/#cell).
+
+### Mobile
+
+N/A
+
+### Globalization
+
+#### Localization of the number
+
+For this we will use `Intl.NumberFormat`, passing in the value of the `lang` design token as the locale.
+
+#### Localization of the unit label
+
+This is more difficult, as there does not seem to be any API for getting a file size unit string for a given locale. Options:
+
+1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
+
+**Pros:** No work for clients.
+
+**Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
+
+2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
+
+**Pros:** Supports any language a client cares to provide translations for.
+
+**Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
+
+I suggest we go with option 1, primarily because I am hesitant to add twelve new label provider strings that clients are expected to localize. I suggest initially supporting the following languages:
```

Comment:
I can't believe I missed that. However, after just a little experimentation, I'm pretty unhappy with some of the formatting and translations.
English:
![image](https://github.com/ni/nimble/assets/7282195/afb4348b-11b9-448f-b95b-034c4f1e31c2)
- No space betwen number and units
- "B" instead of "byte" or "bytes"
- "BB" instead of "GB"

French:
![image](https://github.com/ni/nimble/assets/7282195/1b111374-4972-4b0c-b9c7-f4fa86c4ca28)
- inconsistent spacing between number and units
- "o" instead of "octet" or "octets"
- "Mdo" instead of "Go"
- "Bno" instead of "To"

German:
![image](https://github.com/ni/nimble/assets/7282195/769df939-424b-477c-885a-46440bb166dd)
- these all seem weird, especially the lack of KB

Spanish
![image](https://github.com/ni/nimble/assets/7282195/de6e5837-1b60-41ff-8d5e-4f07e5f135ee)
- no GB, and others seem questionable

I could go on, but in general, I have a lot of doubts about letting `Intl.NumberFormat` handle the unit translation.

Reply:
I have added options 3 and 4 that are variations on using `Intl.NumberFormat` just for unit translation. Both require giving up support for the KiB-type unit labels in one way or another.

Because of the cons of the `Intl.NumberFormat` approaches, I'm now advocating for option 2. I'm waiting to hear others' votes, though.

Reply:
I'm also liking option 2 best right now.

Reply:
One downside of option 2 is that we still don't have a good story for localizing Nimble strings in Blazor apps. That's been less noticeable for tooltips and accessible labels but might become more noticeable with visible column text.

Reply:
Gah... I think I'm leaning towards option 1...

Option 2 pushes a fair bit of integration work on everyone who tries to use the column. How much benefit are they really getting from the column if they have to juggle and possibly mess-up or be inconsistent about digital vs decimal translations across languages. Seems extra unsatisfying work. I can imagine wanting to use a theme-provider to override labels for different increment / decrement buttons within a page, but I can't really imagine a scenario where that flexibility is good here.

And it's a bummer that the standards groups are pushing back on the EIC formats of mebibytes, etc from following these threads: https://github.com/tc39/ecma402/issues/32#issuecomment-468172074

I think we should support those for the accuracy.. I don't think anyone would actually be completely lost by us using KiB and MiB for digital. Think they'll figure it out and maybe learn something. Would be nice to be able to help build some consistency for units at a T&M company, maybe this can be our push for that :) (related wonder if the NI brand guidelines cover any of this)

Reply:
Devtools research on trying to be consistent: https://goo.gle/devtools-si 

and the relevant xkcd lol
![image](https://github.com/ni/nimble/assets/1588923/66e219a6-d4d1-4f00-a291-bf7e9a00eb4a)

Reply:
Another meme, this one from reddit (we did the full curve!):
![image](https://github.com/ni/nimble/assets/1588923/abf8f84c-38e1-4101-8371-882e6f6d0608)

The comments are interesting: https://www.reddit.com/r/ProgrammerHumor/comments/17r2bqx/aretheystupid/

Someone from Azure discussed there approach for defining column units in a table: https://www.reddit.com/r/ProgrammerHumor/comments/17r2bqx/aretheystupid/k8h2fw7/?context=10000

Direct link to Azure column handling docs: https://github.com/Azure/portaldocs/blob/main/portal-sdk/generated/declarative-assets.md#custom-columns